Source/DafnyLanguageServer/Language/DiagnosticPublisher.cs

```diff
@@ -14,13 +14,17 @@ public class DiagnosticPublisher : IDiagnosticPublisher {
     }
 
     public void PublishDiagnostics(DafnyDocument document) {
+      if (document.LoadCanceled) {
+        // Since VSCode correctly shifts the error locations, the error locations are not migrated
```

Comment:
I tried to rephrase the expression. I left VSCode note since I do not know if it's part of the LSP specification that language clients should try to re-locate the diagnostics after document changes. If you still prefer to leave it out, I'll remove it.

Reply:
> I do not know if it's part of the LSP specification that language clients should try to re-locate the diagnostics after document changes

Well, either a client migrates the diagnostics upon a change, or it has to hide them until new diagnostics come in which would give a flickering effect, right? I think all clients are forced to migrate diagnostics to prevent that.

Reply:
Makes sense to me. I've removed the VSCode specific note.